optional_plugins/varianter_cit/avocado_varianter_cit/__init__.py

```diff
@@ -0,0 +1,326 @@
+        # Order of combinations
+        self.order = order
+        self.hash_table = {}
+
+    def combine(self):
+        """
+        Computes the combination of parameters
+
+        :returns: headers (list of parameters keys) and combinations (list of
+                  dictionaries. Each dictionary represents a combination of
+                  parameters.
+        :rtype: tuple
+        """
+        self.create_interaction_hash_table()
+        final_list = self.create_final_list()
+
+        headers = [item[0] for item in self.parameters]
+        result = [[self.parameters[i][1][combination[i]]
+                   for i in range(len(combination))]
+                  for combination in final_list]
+        combinations = []
+        for combination in result:
+            combinations.append(dict(zip(headers, combination)))
+
+        return headers, combinations
+
+    def create_interaction_hash_table(self):
+        for c in itertools.combinations(range(len(self.parameters_length)), self.order):
+            self.hash_table[c] = self.get_interaction(c)
+
+    def create_final_list(self):
+        final_list = []
+        while len(self.hash_table) != 0:
+            iterations = 0
+            previous_test_case = []
+            previous_remove_list = {}
+            previous_weight = 0
+
+            while iterations < self.MAX_ITERATIONS:
+                max_width = len(self.hash_table)
+                matrix = self.create_random_matrix()
+                remove_list = {}
+                for i in matrix:
+                    width = self.get_weight(i, remove_list)
+                    if width == 0 or width <= previous_weight:
+                        remove_list.clear()
+                        continue
+                    elif width == max_width:
+                        final_list.append(i)
+                        self.remove_from_hash_table(remove_list)
+                        previous_test_case = []
+                        previous_remove_list = {}
+                        continue
+                    elif width > previous_weight:
+                        previous_weight = width
+                        previous_test_case = i
+                        previous_remove_list = dict(remove_list)
+                        remove_list.clear()
+                iterations += 1
+            if len(previous_test_case) != 0:
+                previous_remove_list = self.neighborhood_search(
+                    previous_test_case, previous_weight,
+                    max_width, previous_remove_list)
+                final_list.append(previous_test_case)
+                self.remove_from_hash_table(previous_remove_list)
+        return final_list
+
+    def get_interaction(self, parameter_combination):
+        parameters_array = []
+        for c in parameter_combination:
+            array = range(self.parameters_length[c])
+            parameters_array.append(array)
+        iterations = {}
```

Comment:
here it probably should be `interactions`

Reply:
I dare to say that the name of the function needs to be changed, and not this variable.  I'll go this way, and then we can re-conciliate that with @bestoun.